kolibri/plugins/document_pdf_render/assets/src/views/index.vue

```diff
@@ -63,48 +75,83 @@
       targetTime() {
         return this.totalPages * 30;
       },
+      documentLoading() {
+        return this.progress !== 1;
+      },
     },
     methods: {
       toggleFullscreen() {
-        if (this.isFullscreen) {
-          if (this.fullscreenAllowed) {
-            ScreenFull.toggle(this.$refs.container);
-          }
-          this.isFullscreen = false;
+        if (this.fullscreenAllowed) {
+          ScreenFull.toggle(this.$refs.container);
         } else {
-          if (this.fullscreenAllowed) {
-            ScreenFull.toggle(this.$refs.container);
-          }
-          this.isFullscreen = true;
+          this.isFullscreen = !this.isFullscreen;
         }
+        // might want to reset pageScale here
+      },
+      zoomIn() {
+        this.scale += 0.1;
       },
-      getPage(pageNum) {
-        return this.pdfDocument.getPage(pageNum);
+      zoomOut() {
+        this.scale -= 0.1;
       },
-      displayPage(pdfPage) {
+      getPage(pageNum, firstRender = false) {
```

Comment:
Probably ditching the firstRenderOption? Or just renaming `setupIntialPageScale` to `setupZoom`